packages/backend/src/models/CatalogModel/CatalogModel.ts

```diff
@@ -432,49 +432,89 @@ export class CatalogModel {
+                                                fieldType,
+                                            ),
+                                    );
+                                },
+                            );
+                        })
                         .andWhere(
                             `${CatalogTableName}.project_uuid`,
                             projectUuid,
                         )
                         .increment('chart_usage', 1),
-            );
+                );
+            }
 
-            const decrementPromises = fieldsToDecrement.map(
-                ({ fieldName, cachedExploreUuid }) =>
+            // Decrement
+            if (fieldsToDecrement.length > 0) {
+                transactions.push(
                     trx(CatalogTableName)
-                        .where(`${CatalogTableName}.name`, fieldName)
-                        .andWhere(
-                            `${CatalogTableName}.cached_explore_uuid`,
-                            cachedExploreUuid,
-                        )
+                        .where((builder) => {
+                            fieldsToDecrement.forEach(
+                                ({
+                                    cachedExploreUuid,
+                                    fieldName,
+                                    fieldType,
+                                }) => {
+                                    void builder.orWhere((orBuilder) =>
+                                        orBuilder
+                                            .where(
+                                                `${CatalogTableName}.cached_explore_uuid`,
+                                                cachedExploreUuid,
+                                            )
+                                            .andWhere(
+                                                `${CatalogTableName}.name`,
+                                                fieldName,
+                                            )
+                                            .andWhere(
+                                                `${CatalogTableName}.field_type`,
+                                                fieldType,
+                                            ),
+                                    );
+                                },
+                            );
+                        })
                         .andWhere(
                             `${CatalogTableName}.project_uuid`,
                             projectUuid,
                         )
-                        .where('chart_usage', '>', 0) // Ensure we don't decrement below 0
+                        .andWhere('chart_usage', '>', 0) // Ensure we don't decrement below 0
                         .decrement('chart_usage', 1),
-            );
+                );
+            }
 
-            await Promise.all([...incrementPromises, ...decrementPromises]);
+            await Promise.all(transactions);
```

Comment:
@joaoviana Updated the code so it runs at most 2 transactions, 1 to increment and one to decrement

here's the resulting sql:

Increment
```sql
update "catalog_search" set "chart_usage" = "chart_usage" + ? where (("catalog_search"."cached_explore_uuid" = ? and "catalog_search"."name" = ? and "catalog_search"."field_type" = ?) or ("catalog_search"."cached_explore_uuid" = ? and "catalog_search"."name" = ? and "catalog_search"."field_type" = ?)) and "catalog_search"."project_uuid" = ?
```

Decrement
```sql
update "catalog_search" set "chart_usage" = "chart_usage" - ? where (("catalog_search"."cached_explore_uuid" = ? and "catalog_search"."name" = ? and "catalog_search"."field_type" = ?) or ("catalog_search"."cached_explore_uuid" = ? and "catalog_search"."name" = ? and "catalog_search"."field_type" = ?)) and "catalog_search"."project_uuid" = ? and "chart_usage" > ?
```

In both cases I added/removed 2 metrics so we could see the `OR` statements